@xen-orchestra/lite/src/views/new-vm/NewVmView.vue

```diff
@@ -0,0 +1,989 @@
+    network: net.interface,
+    mac: net.macAddress,
+  })),
+  CPUs: vmState.vCPU,
+  VCPUsMax: vmState.VCPUs_max,
+  name_description: vmState.vm_description,
+  memory: vmState.ram,
+  autoPoweron: vmState.auto_power,
+  bootAfterCreate: vmState.boot_vm,
+  copyHostBiosStrings: vmState.boot_firmware !== 'uefi' && !templateHasBiosStrings.value && vmState.copyHostBiosStrings,
+  hvmBootFirmware: vmState.boot_firmware,
+  tags: vmState.tags.split(',').map(tag => tag.trim()),
+  cloudConfig: '',
+}))
+
+const xapi = useXenApiStore().getXapi()
+
+const createVM = async () => {
+  const templateRef = vmCreationParams.value.template
+  const newVmName = vmCreationParams.value.name_label
+  const selectedVdiRef = vmCreationParams.value.installRepository
+  const newVDIs = vmCreationParams.value.VDIs
+  const existingDisks = vmCreationParams.value.existingDisks
+
+  const $defer = []
+
+  if (!templateRef) {
+    console.error('Error : templateRef is undefined or invalid.')
+    return
+  }
+
+  isLoading.value = true
+
+  try {
+    const vmRef = vmCreationParams.value.clone
+      ? await xapi.vm.clone({ [templateRef]: newVmName })
+      : await xapi.vm.copy({ [templateRef]: newVmName }, '' as XenApiSr['$ref'])
+
+    $defer.push(() => xapi.vm.delete(vmRef))
+
+    // Removes disks from the provision XML, we will create them by ourselves.
+    await xapi.vm.removeFromOtherConfig(vmRef, 'disks')
+
+    // Inspects the disk configuration contained within the VM's other_config,
+    // creates VDIs and VBDs and then executes any applicable post-install script.
+    await xapi.vm.provision(vmRef)
+
+    // INSTALL SETTINGS
+    const vm = await xapi.getField<XenApiVm>('VM', vmRef[0], 'record')
+
+    const installMethod = vmState.selectedVdi ? 'cd' : 'network'
+
+    // TODO maybe improve this part
+    if (vm.domain_type === 'hvm') {
+      if ((newVDIs.length === 0 && existingDisks.length === 0) || installMethod === 'network') {
+        const { order } = vm.HVM_boot_params
+        await xapi.call('VM.set_HVM_boot_params', [vmRef[0], { order: order ? 'n' + order.replace('n', '') : 'ncd' }])
+      }
+    } else {
+      if (vm.PV_bootloader === 'eliloader') {
+        if (installMethod === 'network') {
+          await xapi.call('VM.set_other_config', [vmRef[0], 'install-repository', undefined])
+        } else if (installMethod === 'cd') {
+          await xapi.call('VM.set_other_config', [vmRef[0], 'install-repository', 'cdrom'])
+        }
+      }
+    }
+
+    if (installMethod === 'cd') {
+      const VBDs = await xapi.getField<XenApiVbd['$ref'][]>('VM', vmRef[0], 'VBDs')
+
+      for (const vbdRef of VBDs) {
+        const type = await xapi.getField<string>('VBD', vbdRef, 'type')
+        if (type === 'CD') {
+          await xapi.vbd.insert(vbdRef, selectedVdiRef)
+          await xapi.vbd.setBootable(vbdRef, true)
+          return
+        }
+      }
+
+      await xapi.vbd.create({
+        vmRefs: vmRef,
```

Comment:
Can be simplified, if renaming `vmRef` to `vmRefs`, which represents better the fact it is an array of refs:
 ```suggestion
         vmRefs,
```